File: src/main/clients/dbus/PowerProfilesClient.ts
import { PowerProfile } from '@commons/models/Platform';
import { Variant } from 'dbus-next';

import { AbstractDbusClient } from './base/AbstractDbusClient';

export class PowerProfilesClient extends AbstractDbusClient {
  constructor() {
    super(
      'system',
      'net.hadess.PowerProfiles',
      '/net/hadess/PowerProfiles',
      'net.hadess.PowerProfiles'
    );
  }

  public async getActiveProfile(): Promise<PowerProfile> {
    return this.getProperty('ActiveProfile') as unknown as PowerProfile;
  }

  public async setActiveProfile(profile: PowerProfile): Promise<void> {
    return this.setProperty('ActiveProfile', new Variant('s', profile));
  }
}

export const powerProfilesClient = new PowerProfilesClient();

File: src/clients/dbus/power_profiles_client.py
from .base.abstract_dbus_client import AbstractDbusClient
from ...models.power_profile import PowerProfile
from ...utils.singleton import singleton


@singleton
class PowerProfilesClient(AbstractDbusClient):
    def __init__(self):
        super().__init__(
            True,
            "net.hadess.PowerProfiles",
            "/net/hadess/PowerProfiles",
            "net.hadess.PowerProfiles",
        )

    @property
    def activeProfile(self) -> PowerProfile:
        return PowerProfile(self.getProperty("ActiveProfile"))

    @activeProfile.setter
    def activeProfile(self, val: PowerProfile) -> None:
        self.setProperty("ActiveProfile", val.value)


power_profile_client = PowerProfilesClient()
